### DISCUSSION TITLE
How am I supposed to use a scoop program in an elevated shell?

### DISCUSSION THREAD

**Discussion by ekawatani** [ScoopInstaller/Scoop#5309] [Help and Questions]
I want to install postgresql via scoop and set it up as a service. After I installed postgresql, I got this notes:

`register it as a service by running 'pg_ctl register -N PostgreSQL' from an elevated shell.`

![image](https://user-images.githubusercontent.com/61330990/209738223-3b769b22-e6e2-4da4-b27b-18d8ed6e49ec.png)

But, when I opened my terminal in admin mode and try to run it, the `pg_ctl` command was not found. How am I supposed to do this?

**Comment by HUMORCE** [comment]
```
❯ scoop install postgresql
Installing '7zip19.00-helper' (19.00) [64bit] from main bucket
7z1900-x64.msi (1.7 MB) [=================================================================================================] 100%
Checking hash of 7z1900-x64.msi ... ok.
Extracting 7z1900-x64.msi ... done.
Linking ~\scoop\apps\7zip19.00-helper\current => ~\scoop\apps\7zip19.00-helper\19.00
Creating shim for '7z1900-helper'.
'7zip19.00-helper' (19.00) was installed successfully!
Installing 'postgresql' (15.1) [64bit] from main bucket
Loading postgresql-15.1-1-windows-x64-binaries.zip from cache
Checking hash of postgresql-15.1-1-windows-x64-binaries.zip ... ok.
Running pre_install script...
Linking ~\scoop\apps\postgresql\current => ~\scoop\apps\postgresql\15.1
Creating shortcut for pgAdmin 4 (pgAdmin4.exe)
Persisting data
Running post_install script...
'postgresql' (15.1) was installed successfully!
Notes
-----
Run run 'pg_ctl start' or 'pg_ctl stop' to start and stop the database or
register it as a service by running 'pg_ctl register -N PostgreSQL' from an elevated shell.
Default superuser login: postgres, password: <blank>
~ took 22s
❯ Get-Command pg_ctl

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     pg_ctl.exe                                         15.0.1.22… C:\Users\humorce\scoop\apps\postgresql\current\bin\p…

~
❯ pg_ctl --version
pg_ctl (PostgreSQL) 15.1
```

Cannot reproduce.

- Close entire Windows Terminal process(this case) and retry.
- Run direct:

```
❯ cd (scoop prefix postgresql)
❯ .\bin\pg_ctl.exe --version
pg_ctl (PostgreSQL) 15.1
```

Troubleshooting:

```
...
"env_add_path": "bin",
...
```

Please check that `...\postgresql\current\bin` is added to `$env:PATH`.

**Reply by ekawatani** [reply]
@HUMORCE  That's not at all what I said. I of course has access to `pg_ctl` inside my regular shell. That works. What I'm saying is that after I install it in non-elevated shell, then I don't have access it in elevated shell. 

The message says `register it as a service by running 'pg_ctl register -N PostgreSQL' from an elevated shell.`, but this appears to be not possible.

**Reply by HUMORCE** [reply]
`sudo pg_ctl <...>`

**Reply by ekawatani** [reply]
@HUMORCE This is Windows, isn't it? I don't think I have sudo.

**Reply by ekawatani** [reply]
@HUMORCE Any ideas?

**Reply by HUMORCE** [reply]
```
scoop install sudo 
scoop install gsudo
```

**Reply by HUMORCE** [reply]
I am not sure what your actual question is, but you mentioned: 

> when I opened my terminal in admin mode and try to run it, the pg_ctl command was not found. How am I supposed to do this?

`pg_ctl` can be found in any session when I reproduce this.

**Reply by ekawatani** [reply]
@HUMORCE I opened PowerShell, run `scoop install postgresql`. Then, I have access to `pg_ctl` inside this shell. This works as expected. But, when I open PowerShell in admin mode, then `pg_ctl` doesn't exist. It seems I had to use `sudo` from the non-elevated shell.

By the way, the user where I'm logged in is not a admin user. (If you are on Windows 11, go to Settings > Accounts > Email & accounts, then you can add a non-admin user. That's how I added mine.) Maybe this has something to do with it.

**Reply by HUMORCE** [reply]
Install `postgresql` in user A(non-admin) and run it through user B(admin)? then you get the prompt: <command> not found (in PATH).

Operations requiring elevated privileges must be performed by admin user.(in this case: creating service)

https://github.com/ScoopInstaller/Scoop/wiki/Global-Installs

**Reply by ekawatani** [reply]
Thanks for clarifying. For now, I decided to do exactly that. I installed using the admin account.
